### ISSUE TITLE
Load Fairseq model with `BARTModel.from_pretrained` using external BPE

### ISSUE THREAD

**Issue by jbingel** [pytorch/fairseq#3585]
#### What is your question?
I've got a BART model that was pretrained with the `fairseq-train` CLI, and that now consists of three files: `model.pt`, `dict.source.txt` and `dict.target.txt`.  I can load the model successfully with `BARTModel.from_pretrained("./path/to/model")`.

**However,** the two dict files only map numbers to numbers, e.g. the lines in `dict.source.txt` just look like
```
620 22459230
673 22176757
616 21469320
```

So, I need some vocabulary/ID mapping from somewhere, and I noticed that the model is elsewhere used with an external BPE vocabulary, provided in a directory that again contains three files: `dict.txt`, `encoder.json`, `vocab.bpe`.

I assume I somehow have to generate BPE encodings from my input before passing it to the model, using the data provided in the BPE directory. **But how, and using which source?** The `dict.txt` again just maps numbers (and has the same number of lines as the `dict.*.txt` files in the model directory, while the `vocab.bpe` has one item per line, where one item is what looks like subword units, though with some formatting I haven't previously seen. Some lines from the file (which is a few hundred lines shorter that the `dict*txt` files):

```
Ġm on
Ġl ast
Ġ2 00
1 0
Ġst ud
u res
```

Are the leading `Ġ` just an equivalent of what is elsewhere often `##`? And what do the spaces mean?

**So, in short, my question is:** How do I enable my model to use the BPE information above, not using the Fairseq CLI, but just the `BARTModel` class?

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 0.10.0
 - PyTorch Version (e.g., 1.0) 1.8.1
 - OS (e.g., Linux): Ubuntu 20.04
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8.5
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by jbingel** [comment]
Update: I noticed that when I just load the model as above (`bart = BARTModel.from_pretrained("./path/to/model")`) and then call `bart.bpe.bpe.decoder`, I get a dictionary that maps IDs to subwords. **But where does this mapping come from?** Since it's not coded into the `dict.*.txt` files, is it encoded in the `model.pt`? Or is it derived from some implicit default in fairseq?

**Comment by pangsg** [comment]
How can I generate the dict.source.txt ? I hava already finetune bart.large on my own dataset, but when I load `bart = BARTModel.from_pretrained("./path/to/model"))`, it shows me that no such file or directory.
